lib/validators/one_favorite_per_owner_validator.rb

```diff
@@ -0,0 +1,16 @@
+module Validators
+  # Validator for Collection model
+  # Validate that only one favorite collection is created per owner for each project
+  class OneFavoritePerOwnerValidator < ActiveModel::Validator
+    def validate(new_record)
+      return unless new_record.owner
+
+      records = new_record.owner.collections
```

Comment:
you can search for favourites here like 
```suggestion
      owners_existing_favorites = new_record.owner.collections.where(favorite: true)
```